src/sbt-test/sbt-lock/staleHash/test

```diff
@@ -0,0 +1,22 @@
+> lock
+
+# Precondition: jackson is inherited from transitiveProject.
+> reload
+> thisProject/checkExistsDependencyClasspath "com.fasterxml.jackson.core" "jackson-core" "2.2.2"
+> thisProject/assertSbtLockHashIsUpToDate
+> thisProject/assertSbtLockHashIsUpToDateInThisBuild
+
+# Update the jackson version in transitiveProject
+$ delete transitiveProject/build.sbt
+$ copy-file transitiveProject/build.sbt.changed.txt transitiveProject/build.sbt
+
+# Stale lock.sbt hash should suppress the dependencyOverrides.
+> reload
+> thisProject/checkExistsDependencyClasspath "com.fasterxml.jackson.core" "jackson-core" "2.2.3"
+> thisProject/checkAbsentDependencyClasspath "com.fasterxml.jackson.core" "jackson-core" "2.2.2"
+
+# hash of thisProject has not changed:
+> thisProject/assertSbtLockHashIsUpToDate
+
+# but a hash of its dependency has changed:
+-> thisProject/assertSbtLockHashIsUpToDateInThisBuild
```

Comment:
There's something surprising about `thisProject/sbtLockHashIsUpToDate.value == true`, but `thisProject/sbtLockHashIsUpToDate in ThisBuild).value == false`. It makes sense if you know that the implementation literally takes a hash of the current project's `libraryDependencies`, but I worry that users may assume `sbtLockHashIsUpToDate` includes transitive dependencies as well. Seems like a usability issue. It would be great if we could figure out how to have the hash include `libraryDependencies` of transitive project dependencies too. Probably out of scope of this PR.

Reply:
I hate `ThisBuild`, it's such an ugly weird hack.

Reply:
Done.